webapp/src/main/new/New.tsx

```diff
@@ -1,49 +1,61 @@
-import { useState, useMemo } from "react";
+import { useState, useMemo, useEffect } from "react";
+import { Shimmer, ShimmerElementType } from "@fluentui/react/lib/Shimmer";
 
 import { QuadratDataEntryForm } from "./QuadratDataEntryForm";
 import { QuadratMetadataEntryForm } from "./QuadratMetadataEntryForm";
-import { Stem } from "../../types";
-import { insertStems } from "./dataService";
 import { columns } from "./QuadratDataEntryForm/columnHeaders";
-import { mockCensusData } from "../../mockData/mockCensusData";
+
 import {
   postValidate,
   PostValidationError,
 } from "../../validation/postValidation";
+import { getCensus, insertCensus } from "./dataService";
+import { Stem } from "../../types";
+import { useStorageContext } from "../../context/storageContext";
+import { useConnectivityContext } from "../../context/connectivityContext";
+import { getAllItems } from "../../helpers/storageHelper";
 
 export const New = () => {
-  const [errors, setErrors] = useState<any>([]);
+  const { latestCensusStore, userInputStore } = useStorageContext();
+  const { isOnline } = useConnectivityContext();
 
-  const hardCodedFormData: Stem[] = [
-    {
-      Subquadrat: 11,
-      Tag: 1,
-      SpCode: "species",
-      DBH: 10,
-      Htmeas: 1.5,
-      Codes: "at",
-      Comments: "",
-    },
-    {
-      Subquadrat: 11,
-      Tag: 2,
-      SpCode: "species",
-      DBH: 10,
-      Htmeas: 1.5,
-      Codes: "at",
-      Comments: "",
-    },
-  ];
-
-  const [formData] = useState(hardCodedFormData);
+  const [errors, setErrors] = useState<any>([]);
+  const [isLoading, setIsLoading] = useState(false);
 
   // Table data has to be memoized for react-table performance
   const columnHeaders = useMemo(() => columns, []);
-  const [data, setData] = useState(useMemo(() => mockCensusData, []));
+  const [data, setData] = useState<Stem[]>([]);
```

Comment:
This probably doesn't matter since we're not gonna be working with a lot of data, but react-table recommends memoizing the data for performance

Reply:
Thanks for sharing the knowledge. I'm not sure how the data (fetching from the api) should be memorized in this case. thoughts?